src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -1784,6 +1791,80 @@ void returnHttp400_when_CancelItemHold_if_StateIsNotItemHold(InnReachTransaction
+    "CANCEL_REQUEST", "FINAL_CHECKIN", "RECALL", "TRANSFER", "OWNER_RENEW"})
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void returnHttp400WhenFinalCheckInItemHoldStateIsNotItemReceivedOrReceiveUnannounced(InnReachTransaction.TransactionState state) {
+    modifyTransactionState(PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, state);
+
+    var responseEntity = testRestTemplate.postForEntity(
+      ITEM_HOLD_FINAL_CHECK_IN_ENDPOINT, null, Error.class,
+      PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, UUID.randomUUID());
+
+    assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+
+    assertNotNull(responseEntity.getBody());
+    assertTrue(responseEntity.getBody().getMessage().contains("Unexpected transaction state"));
+  }
+
+  @ParameterizedTest
+  @EnumSource(names = {"ITEM_RECEIVED", "RECEIVE_UNANNOUNCED"})
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void testItemHoldFinalCheckInWhenLoanIsClosed(InnReachTransaction.TransactionState state) {
+    var loanStatus = new LoanStatus()
+      .name("closed");
+    var loan = new LoanDTO()
+      .action("checkedin")
+      .status(loanStatus);
+
+    modifyTransactionState(PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, state);
+
+    when(circulationClient.findLoan(any())).thenReturn(Optional.ofNullable(loan));
+
+    var responseEntity = testRestTemplate.postForEntity(
+      ITEM_HOLD_FINAL_CHECK_IN_ENDPOINT, null, void.class,
+      PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, UUID.randomUUID()
+    );
+
+    assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+
+    var transaction = repository.fetchOneById(PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID);
+    assertEquals(FINAL_CHECKIN, transaction.get().getState());
+
+    verify(actionNotifier).reportFinalCheckIn(any());
+    verify(innReachExternalService).postInnReachApi(anyString(), anyString());
+  }
+
+  @ParameterizedTest
+  @EnumSource(names = {"ITEM_RECEIVED", "RECEIVE_UNANNOUNCED"})
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void testItemHoldFinalCheckInWhenLoanIsOpen(InnReachTransaction.TransactionState state) {
+
+    modifyTransactionState(PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, state);
+
+    when(circulationClient.findLoan(any())).thenReturn(Optional.of(new LoanDTO()));
+
+    var responseEntity = testRestTemplate.postForEntity(
+      ITEM_HOLD_FINAL_CHECK_IN_ENDPOINT, null, void.class,
+      PRE_POPULATED_ITEM_HOLD_TRANSACTION_ID, UUID.randomUUID()
+    );
+
+    assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+
+    verify(loanService).checkInItem(any(), any());
```

Comment:
`verify(circulationClient).checkInByBarcode` instead. No need to add one more bean just for this purpose

Reply:
fixed